lighthouse-treemap/app/src/main.js

```diff
@@ -202,30 +251,49 @@ class TreemapViewer {
 }
 
 /**
- * @param {number} scriptsResourceSize
+ * @param {LH.Treemap.ViewMode[]} viewModes
  */
-function renderViewModeOptions(scriptsResourceSize) {
-  const viewModesEl = TreemapUtil.find('.lh-modes');
-  viewModesEl.innerHTML = '';
-
+function renderViewModeButtons(viewModes) {
   /**
-   * @param {string} label
-   * @param {number} bytes
+   * @param {LH.Treemap.ViewMode} viewMode
    */
-  function makeViewMode(label, bytes) {
-    const modesEl = TreemapUtil.createChildOf(viewModesEl, 'div', 'view-mode');
-    modesEl.classList.add('view-mode--active');
-
-    TreemapUtil.createChildOf(modesEl, 'span').textContent = label;
-    TreemapUtil.createChildOf(modesEl, 'span', 'lh-text-dim').textContent =
-    ` (${TreemapUtil.formatBytes(bytes)})`;
+  function render(viewMode) {
+    const viewModeEl = TreemapUtil.createChildOf(viewModesEl, 'div', 'view-mode');
+    viewModeEl.id = `view-mode--${viewMode.id}`;
+
+    const labelEl =
+      TreemapUtil.createChildOf(viewModeEl, 'label', 'view-mode__label');
+    TreemapUtil.createChildOf(labelEl, 'span').textContent = viewMode.label;
+    TreemapUtil.createChildOf(labelEl, 'span', 'lh-text-dim').textContent =
+      ` (${viewMode.subLabel})`;
+
+    const inputEl = TreemapUtil.createChildOf(labelEl, 'input', 'view-mode__button', {
+      type: 'radio',
+      name: 'view-mode',
+    });
 
-    modesEl.addEventListener('click', () => {
+    inputEl.addEventListener('click', () => {
+      treemapViewer.currentViewMode = viewMode;
       treemapViewer.show();
```

Comment:
so `show` is like `rerender`?

Reply:
combined into just a single `render` fn.